### DISCUSSION TITLE
How can I run the indexing and searching in memory instead of on disk？

### DISCUSSION THREAD

**Discussion by Wenze7** [castorini/pyserini#1382] [Q&A]

**Comment by lintool** [comment]
Hi @Wenze7 - this is an ill-formed question. What exactly are you trying to do? Dense/spare already try to make effective use of memory, but in different ways.

**Reply by Wenze7** [reply]
Sorry. According to my guess, unlike faiss, sparse's main bottleneck is hard disk. Can I run sparse in memory in the same way as the faiss？

**Reply by lintool** [reply]
Nope, it doesn't work like that. There's an in-memory option in Anserini on the Lucene end, but based on my experience, it doesn't increase retrieval speed by much. The standard way to increase speed is to partition the dataset and search each partition in parallel.

**Reply by TylerRankin** [reply]
Im running the pyserini LuceneSearcher on azure web app, so there is latency when loading in indexes from azure storage. Is there anyway to pre-load in all the indexes I need ? Thanks !
